src/queue/memoryQueue.ts

```
@@ -0,0 +1,65 @@
+import { QueueId, QueueInterface } from "./queue";
+import { DSNPMessage } from "../messages/messages";
+
+export const IdDoesNotExist = new Error("No message matching the given ID exists.");
```

Should this be exported?

---

It's used in tests, but I don't have any other good reason to export it.

I suppose a user might want to try-catch the `dequeue` method and ignore this error in particular while still raising exceptions for memory issues, but they could also accomplish that by just implementing their own queue.

---

Wouldn't we want to have custom error classes then?

---

Removed the export